### PR TITLE
Added vitest for VolunteerContainer

### PR DESCRIPTION
**Pull request by NishantSinghhhhh**
PR Title: Refactor VolunteerContainer Component Tests: Migrate from Jest to Vitest
Issue Number:
Fixes: #2548

Did you add tests for your changes?
Yes

Snapshots/Videos:

[Screencast from 2024-12-21 01-57-16.webm](https://github.com/user-attachments/assets/eb0262bb-b0bf-4691-acbb-1b78f6641f1a)

Summary:
This PR refactors the test suite for the VolunteerContainer component by migrating from Jest to Vitest, in alignment with the project's updated testing framework. The following changes were made:

Updated the testing configuration to be compatible with Vitest.
Refactored all test files related to VolunteerContainer to use Vitest's syntax and features.
Consolidated mock definitions to reduce redundancy across test cases for improved maintainability.

## Summary by CodeRabbit

- **Tests**
	- Updated unit tests for the `VolunteerContainer` component to enhance test structure and parameter handling.
	- Transitioned testing framework from `jest` to `vitest`.
	- Improved mocking of route parameters and asynchronous handling in test cases.
	- Added `data-testid` attribute to the title span element in the `VolunteerContainer` for better testability.

### CODE REVIEW

**Comment by NishantSinghhhhh** [comment]
@varshith257 , can you please review

**Comment by NishantSinghhhhh** [comment]
@varshith257  @palisadoes , can you please review

**Comment by NishantSinghhhhh** [comment]
@varshith257 ,@ pranshugupta54  can you please review this PR

**Comment by NishantSinghhhhh** [comment]
> @NishantSinghhhhh Can you have a look at the coverage report. Although changes seems fine this needs to be in coverage report

Okay sure , will commit the changes by evening

File: src/screens/EventVolunteers/VolunteerContainer.spec.tsx [hunk 1]
```diff
@@ -12,19 +12,41 @@ import userEvent from '@testing-library/user-event';
...
     <MockedProvider addTypename={false} link={link1}>
-      <MemoryRouter initialEntries={['/event/orgId/eventId']}>
+      <MemoryRouter initialEntries={[initialPath]}>
         <Provider store={store}>
           <LocalizationProvider>
             <I18nextProvider i18n={i18n}>
               <Routes>
                 <Route
-                  path="/event/:orgId/:eventId"
+                  path="/org/:orgId/event/:eventId"
```
Comment (palisadoes) [inline src/screens/EventVolunteers/VolunteerContainer.spec.tsx:49]: Why was this changed?

File: src/screens/EventVolunteers/VolunteerContainer.spec.tsx [hunk 2]
```diff
@@ -12,19 +12,41 @@ import userEvent from '@testing-library/user-event';
...
+  };
+});
+
+const renderVolunteerContainer = (
+  initialPath = '/org/orgId/event/eventId',
+): RenderResult => {
   return render(
     <MockedProvider addTypename={false} link={link1}>
-      <MemoryRouter initialEntries={['/event/orgId/eventId']}>
+      <MemoryRouter initialEntries={[initialPath]}>
```
Comment (palisadoes) [inline src/screens/EventVolunteers/VolunteerContainer.spec.tsx:43]: Why was this changed?

File: src/screens/EventVolunteers/VolunteerContainer.spec.tsx [hunk 3]
```diff
@@ -41,54 +63,37 @@ const renderVolunteerContainer = (): RenderResult => {
...
   });
 
   it('should redirect to fallback URL if URL params are undefined', async () => {
-    render(
-      <MockedProvider addTypename={false} link={link1}>
-        <MemoryRouter initialEntries={['/event/']}>
-          <Provider store={store}>
-            <I18nextProvider i18n={i18n}>
-              <Routes>
-                <Route path="/event/" element={<VolunteerContainer />} />
```
Comment (palisadoes) [inline src/screens/EventVolunteers/VolunteerContainer.spec.tsx:62]: Why was this deleted?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by palisadoes** [comment]
@NishantSinghhhhh You haven't addressed my questions

**Comment by NishantSinghhhhh** [comment]
> @NishantSinghhhhh You haven't addressed my questions

Okay @palisadoes , I am 

> @NishantSinghhhhh You haven't addressed my questions

Have added all the changes you requested

File: src/screens/EventVolunteers/VolunteerContainer.spec.tsx (hunk 3 above)
Comment (palisadoes) [reply src/screens/EventVolunteers/VolunteerContainer.spec.tsx]: You haven't answered this question

**Review by palisadoes — APPROVED** [review]

**Pull request merged**